### PR TITLE
Fix #744 - Stop overwriting class docblock with '' triggered by ::class

### LINKED ISSUE #744: StaticReflectionParser is dumb, assumes Class::class references are class level
If I use the below PHP class structure, then parsing the annotations will always return an empty class level annotation.  The current HEAD of https://www.drupal.org/project/migrate_source_csv is broken right now using this pattern. It is a sneaky breakage, because Unit tests pass. It is only when I run integration tests that is breaks.

See https://www.drupal.org/node/2818891

``` php
/**
 * Source for CSV.
 *
 * @MigrateSource(
 *   id = "csv"
 * )
 */
class CSV extends SourcePluginBase {

  protected $fileClass = '';

  /**
   * {@inheritdoc}
   */
  public function __construct(array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration) {
    $this->fileClass = empty($configuration['file_class']) ? CSVFileObject::class : $configuration['file_class'];
  }
```

### PR DESCRIPTION
**Pull request by Sam152**

### CODE REVIEW

File: lib/Doctrine/Common/Reflection/StaticReflectionParser.php [hunk 1]
```diff
@@ -153,8 +154,10 @@ protected function parse()
                     $docComment = $token[1];
                     break;
                 case T_CLASS:
-                    $this->docComment['class'] = $docComment;
-                    $docComment = '';
+                    if ($last_token !== T_PAAMAYIM_NEKUDOTAYIM) {
```
Comment (heddn) [inline lib/Doctrine/Common/Reflection/StaticReflectionParser.php]: Would it make sense to add a comment to this value? It isn't really clear why or what the the value equates too.

Comment (Sam152) [inline lib/Doctrine/Common/Reflection/StaticReflectionParser.php]: Personally I think the constant is a good enough indicator on it's own. It's pretty easy to lookup if you are unfamiliar with it.

Comment (Ocramius) [inline lib/Doctrine/Common/Reflection/StaticReflectionParser.php]: See https://externals.io/message/99753#99770

File: tests/Doctrine/Tests/Common/Reflection/ExampleAnnotationClass.php [hunk 2]
```diff
@@ -11,4 +11,11 @@
  */
 class ExampleAnnotationClass {
 
+  const foo = \stdClass::class;
+
+  /**
+   * Example with comment.
+   */
+  const bar = \stdClass::class;
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Reflection/ExampleAnnotationClass.php]: @Sam152 these additional symbols should probably be in a separate class/test - try not modifying existing test assets, please

Comment (Sam152) [inline tests/Doctrine/Tests/Common/Reflection/ExampleAnnotationClass.php]: I've added a param to the data provider for the existing annotation test and added a new class for the new test case.

**Comment by Ocramius** [comment]
@Sam152 that looks much better, thanks!

File: lib/Doctrine/Common/Reflection/StaticReflectionParser.php [hunk 3]
```diff
@@ -144,6 +144,7 @@ protected function parse()
         }
         $tokenParser = new TokenParser($contents);
         $docComment = '';
+        $last_token = FALSE;
```
Comment (Ocramius) [inline lib/Doctrine/Common/Reflection/StaticReflectionParser.php]: Just a reminder: lowercase `false` please ;-)

I'm changing it on merge anyway.

**Comment by Ocramius** [comment]
Merged, thanks @Sam152!

**Comment by Ocramius** [comment]
Backported to `2.7` via doctrine/common@6b5b6a700eed1faf7eb76b4eae61b82d915ae6c8

**Comment by Sam152** [comment]
Thanks @Ocramius!

**Pull request merged**